commons-kafka/src/main/scala/com/avsystem/commons/kafka/VersionedSerializer.scala

```diff
@@ -0,0 +1,14 @@
+package com.avsystem.commons
+package kafka
+
+import com.avsystem.commons.kafka.exceptions.UnsupportedVersionEvent
+
+class VersionedSerializer[T](version: Byte)(implicit registry: SerdeRegistry[T]) extends AbstractSerializer[T] {
+  require(registry.isSupported(version), s"Version: $version isn't supported by registry. Please add it first.")
+  override def serialize(topic: String, data: T): Array[Byte] = {
+    registry(version) match {
+      case Some(serde: KafkaSerde[T]) => version +: serde.serializer().serialize(topic, data)
```

Comment:
Ugh, that's going to copy the entire array... can we avoid it?

Reply:
I decided to rewrite KafkaSerde and now it works without copying the array. What do you think?

Reply:
Cool, although I agree with @mkej comments about this implementation.